ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx

```diff
@@ -117,15 +117,31 @@ export const PercentageAndAmountChange = ({
 
   const formattedValue = formatValue(balanceChange === 0 ? 0 : value, true);
 
-  const formattedValuePrice = isValidAmount(balanceChange)
-    ? `${(balanceChange as number) >= 0 ? '+' : ''}${Intl.NumberFormat(locale, {
-        notation: 'compact',
-        compactDisplay: 'short',
+  let formattedValuePrice = '';
+  if (isValidAmount(balanceChange)) {
+    formattedValuePrice = (balanceChange as number) >= 0 ? '+' : '';
+
+    const options = {
+      notation: 'compact',
+      compactDisplay: 'short',
+      maximumFractionDigits: 2,
+    } as const;
+
+    try {
+      // For currencies compliant with ISO 4217 Standard
+      formattedValuePrice += `${Intl.NumberFormat(locale, {
+        ...options,
         style: 'currency',
         currency: fiatCurrency,
-        maximumFractionDigits: 2,
-      }).format(balanceChange as number)} `
-    : '';
+      }).format(balanceChange as number)} `;
+    } catch {
+      // Non-standard Currency Codes
+      formattedValuePrice += `${Intl.NumberFormat(locale, {
```

Comment:
could we add a unit test to ensure that the code path through the catch block is tested ?